spec/unit/parser/functions/tagged_spec.rb

```diff
@@ -22,4 +22,20 @@
       @scope.function_tagged(['one', 'two'])
     end.to raise_error(Puppet::ParseError, /is only available when compiling a catalog/)
   end
+
+  it 'should be case-insensitive' do
+    resource = Puppet::Parser::Resource.new(:file, "/file", :scope => @scope)
+    @scope.expects(:resource).returns resource
+    @scope.function_tag ["one"]
+
+    expect(@scope.function_tagged(['One'])).to eq(true)
+  end
+
+  it 'should check if all specified tags are included' do
+    resource = Puppet::Parser::Resource.new(:file, "/file", :scope => @scope)
+    @scope.expects(:resource).returns resource
```

Comment:
I believe these tests are both failing because we want `stubs` here rather than `expects`. Where `expects` expects the method to be called exactly once, or a specified set of times, `stubs` will accept the method any number of times. Once I made that change, these tests passed for me.

Reply:
Thanks for suggestion @melissa, tests are indeed passing now.